src/actions/attack.cpp

```diff
@@ -1586,7 +1586,7 @@ void attack_unit_and_advance(const map_location& attacker,
 int under_leadership(const unit &u, const map_location& loc, const_attack_ptr weapon, const_attack_ptr opp_weapon)
 {
 	unit_ability_list abil = u.get_abilities_weapons("leadership", loc, weapon, opp_weapon);
-	unit_abilities::effect leader_effect(abil, 0, false);
+	unit_abilities::effect leader_effect(abil, 0, false, const_attack_ptr() , true);
```

Comment:
I think all the lines in this PR should be consistent about using the same choice of `const_attack_ptr()`, `{}` or `nullptr` for the first optional argument.

Reply:
it's done